integration/k8s_conformance_test.go

```diff
@@ -0,0 +1,200 @@
+package integration
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/rs/zerolog/log"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	"github.com/traefik/traefik/v3/integration/try"
+	"github.com/traefik/traefik/v3/pkg/version"
+	"gopkg.in/yaml.v3"
+	ktypes "k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/sets"
+	kclientset "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	gatev1 "sigs.k8s.io/gateway-api/apis/v1"
+	gatev1alpha2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
+	gatev1beta1 "sigs.k8s.io/gateway-api/apis/v1beta1"
+	conformanceV1alpha1 "sigs.k8s.io/gateway-api/conformance/apis/v1alpha1"
+	"sigs.k8s.io/gateway-api/conformance/tests"
+	"sigs.k8s.io/gateway-api/conformance/utils/config"
+	ksuite "sigs.k8s.io/gateway-api/conformance/utils/suite"
+)
+
+// K8sConformanceSuite tests suite.
+type K8sConformanceSuite struct{ BaseSuite }
+
+func TestK8sConformanceSuite(t *testing.T) {
+	suite.Run(t, new(K8sConformanceSuite))
+}
+
+func (s *K8sConformanceSuite) SetupSuite() {
+	s.BaseSuite.SetupSuite()
+
+	s.createComposeProject("k8s")
+	s.composeUp()
+
+	abs, err := filepath.Abs("./fixtures/k8s/config.skip/kubeconfig.yaml")
+	require.NoError(s.T(), err)
+
+	err = try.Do(60*time.Second, func() error {
+		_, err := os.Stat(abs)
+		return err
+	})
+	require.NoError(s.T(), err)
+
+	data, err := os.ReadFile(abs)
+	require.NoError(s.T(), err)
+
+	content := strings.ReplaceAll(string(data), "https://server:6443", fmt.Sprintf("https://%s", net.JoinHostPort(s.getComposeServiceIP("server"), "6443")))
+
+	err = os.WriteFile(abs, []byte(content), 0o644)
+	require.NoError(s.T(), err)
+
+	err = os.Setenv("KUBECONFIG", abs)
+	require.NoError(s.T(), err)
+}
+
+func (s *K8sConformanceSuite) TearDownSuite() {
+	s.BaseSuite.TearDownSuite()
+
+	generatedFiles := []string{
+		"./fixtures/k8s/config.skip/kubeconfig.yaml",
+		"./fixtures/k8s/config.skip/k3s.log",
+		"./fixtures/k8s/coredns.yaml",
```

Comment:
I'm getting errors when deleting `coredns.yaml` and `traefik.yaml`. Could you remove those?

```
{"level":"warn","error":"remove ./fixtures/k8s/coredns.yaml: no such file or directory","time":"2024-01-19T09:11:49+01:00"}
{"level":"warn","error":"remove ./fixtures/k8s/traefik.yaml: no such file or directory","time":"2024-01-19T09:11:49+01:00"}
```